### DISCUSSION TITLE
How to connect use-pagination hook with <Pagination /> component?

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#3739] [Q&A]
### What package has an issue

@mantine/core

### Describe the bug

How to connect use-pagination hook with `<Pagination />` component?

Want to get control over the existing Mantine `<Pagination />` component but also be able to trigger `pagination.setPage(XXXX)`, `pagination.next()`, etc. 

An example would be great.

### What version of @mantine/hooks page do you have in package.json?

6.0.0

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
You cannot do that. Pagination component uses usePagination hook under the hood. If you want to use usePagination hook, then you will need to build a custom pagination component.

**Reply by atumas-bananamilk** [reply]
So there's no control over the Pagination component to switch between pages?
The problem is when using Pagination component with `useSWR` hook, it caches the pages, so a non-existent page is still shown for a brief moment. Want to programmatically move the user from the last page to the NEW last page without showing the previously-existing last page.

**Reply by rtivital** [reply]
You set current page on your side in controlled Pagination component – https://mantine.dev/core/pagination/#controlled
